### ISSUE TITLE
CI Test Errors FiniteDuration antisymmetry

### ISSUE THREAD

**Issue by Slakah** [typelevel/cats#3734]
Some CI failures on Scala 3.

CI failure from https://github.com/typelevel/cats/pull/3727 :
<details>
  <summary>Click to expand!</summary>

```
failing seed for order.antisymmetry eq is tJVxJ6pgwg7UrNv9sWAqwY4eokyiECpCf8K9kK8SQhC=
failing seed for order.antisymmetry eq is tJVxJ6pgwg7UrNv9sWAqwY4eokyiECpCf8K9kK8SQhC=
==> X cats.kernel.laws.Tests.Order[FiniteDuration]: order.antisymmetry eq  0.046s munit.FailException: /home/runner/work/cats/cats/kernel-laws/shared/src/test/scala/cats/kernel/laws/LawTests.scala:201
200:  checkAll("Order[Duration]", OrderTests[Duration].order)
201:  checkAll("Order[FiniteDuration]", OrderTests[FiniteDuration].order)
202:  checkAll("Order[UUID]", OrderTests[UUID].order)
```
</details>

CI failure from https://github.com/typelevel/cats/pull/3730 :

<details>
  <summary>Click to expand!</summary>

```
failing seed for hash.antisymmetry eq is tJ9I1dkawrvX2DEc6NGtz5qNQw96EBLm1nV0-XvgFmA=
==> X cats.kernel.laws.Tests.Hash[FiniteDuration]: hash.antisymmetry eq  0.028s munit.FailException: /home/runner/work/cats/cats/kernel-laws/shared/src/test/scala/cats/kernel/laws/LawTests.scala:299
298:  checkAll("Hash[Duration]", HashTests[Duration].hash)
299:  checkAll("Hash[FiniteDuration]", HashTests[FiniteDuration].hash)
300:

Failing seed: 5MC5_eL2QrqZQkkWTTuITf7k9M0Wy1b5-RAvsNJZxHO=
You can reproduce this failure by adding the following override to your suite:

  override val scalaCheckInitialSeed = "5MC5_eL2QrqZQkkWTTuITf7k9M0Wy1b5-RAvsNJZxHO="

Falsified after 62 passed tests.
> Labels of failing property: 
Expected: true
Received: false
> ARG_0: -3888 hours
> ARG_1: -162 days
> ARG_2: org.scalacheck.GenArities$$Lambda$458/12131917@1f846122

```

</details>

**Comment by rossabaker** [comment]
Thanks for capturing these!  Note that in both cases, the arguments are equal, but have different units:

```scala
@ 114.days === 2736.hours 
res12: Boolean = true

@ -162.days === -3888.hours 
res13: Boolean = true
```

That suggests a problem with the `Cogen` instance, whereupon we find https://github.com/typelevel/cats/pull/3470, and its reversion in https://github.com/typelevel/cats/pull/3694.

Does the plot thicken with Scala 3?

**Comment by larsrh** [comment]
@TimWSpence also observed this in #3762. I'm tempted to agree that either `antiSymmetryEq` should go, or the `Cogen` instance for `FiniteDuration` is borked.

**Comment by larsrh** [comment]
Fixed by #3772.

**Comment by TimWSpence** [comment]
> @TimWSpence also observed this in #3762. I'm tempted to agree that either `antiSymmetryEq` should go, or the `Cogen` instance for `FiniteDuration` is borked.



> Thanks for capturing these! Note that in both cases, the arguments are equal, but have different units:
> 
> ```scala
> @ 114.days === 2736.hours 
> res12: Boolean = true
> 
> @ -162.days === -3888.hours 
> res13: Boolean = true
> ```
> 
> That suggests a problem with the `Cogen` instance, whereupon we find #3470, and its reversion in #3694.
> 
> Does the plot thicken with Scala 3?

@rossabaker What difference does scala 3 make sorry? The law seems somewhat fundamentally broken to me - if you have a non-trivial `Eq` instance like `FiniteDuration` then you can write perfectly reasonable functions that aren't well defined wrt `Eq` (ie `x == y` does not imply `f x == f y`) eg by pattern-matching on the time unit like in the `Cogen` instance above
